cassandra-scheduler/src/main/java/io/mesosphere/mesos/frameworks/cassandra/CassandraScheduler.java

```diff
@@ -240,196 +236,269 @@ public void error(final SchedulerDriver driver, final String message) {
-                        .setSlaveId(offer.getSlaveId())
-                        .setData(ByteString.copyFrom(taskDetails.toByteArray()))
-                        .addAllResources(newArrayList(
-                            cpu(0.1),
-                            mem(16),
-                            disk(16)
-                        ))
-                        .setExecutor(info)
-                        .build();
-                    LOGGER.debug(marker, "Launching CASSANDRA_NODE_HEALTH_CHECK task", protoToString(task));
-                    driver.launchTasks(newArrayList(offer.getId()), newArrayList(task));
-                    superTasks.add(new SuperTask(offer.getHostname(), task, info, taskDetails));
+                    } else
+                        executorMetadata.notLaunched();
+                } else if (executorMetadata.isLaunched()) {
+                    // TODO this state is reached after the C* daemon has been launched but before we recognize it as running.
+
+                    executorMetadata.setRunning();
+
+                    cluster.nodeRunStateUpdate();
+                }
+            }
+
+            if (executorMetadata.isRunning()) {
+                if (cluster.shouldRunHealthCheck(executorID)) {
+                    submitHealthCheck(marker, driver, offer, executorMetadata);
+                    offerUsed = true;
+                } else if (cluster.shouldGetRepairStatusOnExecutor(executorID)) {
+                    submitRepairStatus(marker, driver, offer, executorMetadata);
+                    offerUsed = true;
+                } else if (cluster.shouldStartRepairOnExecutor(executorID)) {
+                    submitStartRepair(marker, driver, offer, executorMetadata);
                     offerUsed = true;
                 }
             }
         }
 
-
-        if (!offerUsed) {
-            LOGGER.trace(marker, "Declining Offer: {}", offer.getId().getValue());
+        if (!offerUsed)
             driver.declineOffer(offer.getId());
-        }
-        LOGGER.trace(marker, "< evaluateOffer(driver : {}, offer : {})", driver, protoToString(offer));
-        return offerUsed;
     }
 
-    private boolean maybeLaunchExecutor(final SchedulerDriver driver, final Offer offer, final Marker marker, final ListMultimap<ExecutorID, SuperTask> tasksByExecutor) {
-        final int ec = tasksByExecutor.size();
-        final FluentIterable<SuperTask> tasksAlreadyOnHost = from(superTasks).filter(SuperTask.hostnameEq(offer.getHostname()));
-        if (ec < numberOfNodes && tasksAlreadyOnHost.isEmpty()) {
-            final ExecutorID executorId = executorId(frameworkName + ".node." + execCounter.getAndIncrement() + ".executor");
-            final ExecutorInfo info = executorInfo(
-                executorId,
-                executorId.getValue(),
-                frameworkName,
+    private void submitRepairStatus(Marker marker, SchedulerDriver driver, Offer offer, ExecutorMetadata executorMetadata) {
```

Comment:
These new task types should be handled in [`CassandraCluster#getTasksForOffer`](https://github.com/mesosphere/cassandra-mesos/blob/rewrite/cassandra-scheduler/src/main/java/io/mesosphere/mesos/frameworks/cassandra/CassandraCluster.java#L208)